akvo/rsr/models/project.py

```diff
@@ -981,13 +981,9 @@ def budget_total(self):
         return Project.objects.budget_total().get(pk=self.pk).budget_total
 
     def has_multiple_budget_currencies(self):
-        budget_items = BudgetItem.objects.filter(project__id=self.pk)
-        num_currencies = len(set([self.currency] + [c.currency if c.currency else self.currency for c in budget_items]))
-
-        if num_currencies > 1:
-            return True
-        else:
-            return False
+        budget_items = self.budget_items.all()
```

Comment:
Looks like the Python iteration is about a hundred times faster, when the `budget_items` have been pre-fetched and there are only a couple of them.